ansible/roles-infra/infra-osp-project-create/tasks/main.yaml

```diff
@@ -1,79 +1,118 @@
-        description: "{{ env_type }} for user {{ student_name }}"
-        enabled: True
-        domain_id: "Default"
-        endpoint_type: public
+  - name: Create project and quota
+    when: osp_project_create
+    block:
+      - name: Create project for user
+        os_project:
+          name: "{{ osp_project_name }}"
+          state: present
+          description: "{{ env_type }} for user {{ student_name }}"
+          enabled: True
+          domain_id: "Default"
+          endpoint_type: public
 
-    - name: Grant access to admin account on new project
-      os_user_role:
-        state: present
-        user: "{{ admin_user }}"
-        role: "admin"
-        project: "{{ osp_project_name }}"
+      - name: Grant access to admin account on new project
+        os_user_role:
+          state: present
+          user: "{{ admin_user }}"
+          role: "admin"
+          project: "{{ osp_project_name }}"
+
+      - name: Set quotas on new project
+        os_quota:
+          name: "{{ osp_project_name }}"
+          instances: "{{ quota_num_instances }}"
+          cores: "{{ quota_num_cores }}"
+          ram: "{{ quota_memory }}" #in MB
+          volumes: "{{ quota_num_volumes }}"
+          gigabytes: "{{ quota_volumes_gigs }}" #volume storage
+          #loadbalancer: #when Octavia is available
+          #pool: #when Octavia is available
+          network: "{{ quota_networks }}"
+          subnet: "{{ quota_subnets }}"
+          router: "{{ quota_routers }}"
+          floatingip: "{{ quota_fip }}"
+          security_group: "{{ quota_sg }}"
+          security_group_rule: "{{ quota_sg_rules }}"
+
+      - name: Create user in new project
+        os_user:
+          state: present
+          name: "{{ guid }}-user"
+          password: "{{ heat_user_password }}"
+          domain: default
+          default_project: "{{ osp_project_name }}"
 
-    - name: Set quotas on new project
-      os_quota:
-        name: "{{ osp_project_name }}"
-        instances: "{{ quota_num_instances }}"
-        cores: "{{ quota_num_cores }}"
-        ram: "{{ quota_memory }}" #in MB
-        volumes: "{{ quota_num_volumes }}"
-        gigabytes: "{{ quota_volumes_gigs }}" #volume storage
-        #loadbalancer: #when Octavia is available
-        #pool: #when Octavia is available
-        network: "{{ quota_networks }}"
-        subnet: "{{ quota_subnets }}"
-        router: "{{ quota_routers }}"
-        floatingip: "{{ quota_fip }}"
-        security_group: "{{ quota_sg }}"
-        security_group_rule: "{{ quota_sg_rules }}"
+      - name: Add member role to user
+        os_user_role:
+          user: "{{ guid }}-user"
+          role: _member_
+          project: "{{ osp_project_name }}"
 
-    - set_fact:
-        heat_user_password_generated: "{{lookup('password', '/dev/null length=12 chars=ascii_letters,digits')}}"
+      - name: Add Swift role to user
+        os_user_role:
+          user: "{{ guid }}-user"
```

Comment:
shouldn't it be `osp_auth_username_member` ?

Reply:
updated